### ISSUE TITLE
Membership defined by DevCon GitHub organisation membership 

### ISSUE THREAD

**Issue by marcelkornblum** [devcon-london/devcon-london.github.io#11]
I'd love the member directory to come from github - as in every DevCon member is added to [this organisation](https://github.com/devcon-london) within different teams. Benefits:
* If the site is static that gives us a data source for members
* We have a way to do member-only SSO if we ever need to build out features
* There's a built-in directory, but also other GH tools like Projects and Discussions, that might be good for having an open structure, especially if the organisation ends up being more active

This is related to #6 and #9

**Comment by grudelsud** [comment]
sounds cool to me. Couple of things I'm thinking on top of this:

- should be easy enough to write a script that goes through the spreadsheet and send invites for all users that are not organisation members
- once we have a members only page, it'd be probably nice to show data coming both from GH and the spreadsheet in one place

**Comment by marcelkornblum** [comment]
What data would the spreadsheet have that we'd want to connect in? I'm mainly asking because of the (small but vocal) resistance to this spreadsheet, as well as the admin burden of keeping it up to date. 

For example, could we add the relevant fields to the Slack profile form and then just use the GH and Slack APIs together if we ever need a combined view? That way everyone is responsible for their own profile...

**Comment by grudelsud** [comment]
> What data would the spreadsheet have that we'd want to connect in? I'm mainly asking because of the (small but vocal) resistance to this spreadsheet, as well as the admin burden of keeping it up to date.

yup, agree. let's try to clarify things a bit here (so it's written black on white in a public post). I think most of us agree with the nature of DevCon being invitation only has been a good things to keep BS and recruiters away. The spreadsheet at this point would be mainly used a recipient for submissions with the following fields:

- joining date
- email address for the google group
- github profile
- inductee / referrer

Using content from GH will "force" people updating their profile, we'll probably have load of crap, but it's worth giving it a go.

Never used the Slack API, but worth keeping it as a second option. I doubt anyone keeps that data up to date, but maybe I'm wrong.

**Comment by marcelkornblum** [comment]
Ah - I'd forgotten about the Google Group (duh). 

So the spreadsheet is really just useful as a membership / M2M join table to connect the IDs to the same person, and with join date?

Of the three systems (GH/Google/Slack), all have APIs but Slack allows custom profile fields. So in theory we could ask everyone to put their detail in there and ditch the spreadsheet, but probably we'd lose data. Alternative would be to keep the spreadsheet but use it exclusively as a DB for an Apps Script back end, with a members-only front end in the website. Which I guess is your suggestion here :D

**Comment by marcelkornblum** [comment]
useful links regarding this: 
https://api.slack.com/docs/deep-linking#open_a_direct_message (in case we make a directory page, we can link directly to slack DM per person)
https://api.slack.com/methods/users.profile.set (in case we did want to drop the sheet in favour of slack custom fields)

Having seen the sheet though, I'm thinking we will need a central data source (and google sheets might be the right thing for the job)
